### PR TITLE
Release/0.3.1 performance fixes

### PR DESCRIPTION
**Pull request by Skejven**
Pull request with performance tests fixes and documentation.

---

I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: documentation/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
 	<parent>
         <groupId>com.cognifide.knotx</groupId>
         <artifactId>knotx-root</artifactId>
-        <version>0.3.0-SNAPSHOT</version>
+        <version>0.3.1</version>
```
Comment (marcinczeczko) [inline documentation/pom.xml]: Shouldn't be snapshot ?

File: knotx-core/knotx-launcher/pom.xml [hunk 2]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>com.cognifide.knotx</groupId>
         <artifactId>knotx-core</artifactId>
-        <version>0.3.0-SNAPSHOT</version>
+        <version>0.3.1</version>
```
Comment (marcinczeczko) [inline knotx-core/knotx-launcher/pom.xml]: SNAPSHOT ?

File: knotx-core/knotx-repository/pom.xml [hunk 3]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>com.cognifide.knotx</groupId>
         <artifactId>knotx-core</artifactId>
-        <version>0.3.0-SNAPSHOT</version>
+        <version>0.3.1</version>
```
Comment (marcinczeczko) [inline knotx-core/knotx-repository/pom.xml]: SNAPSHOT ?

File: knotx-core/knotx-server/pom.xml [hunk 4]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>com.cognifide.knotx</groupId>
         <artifactId>knotx-core</artifactId>
-        <version>0.3.0-SNAPSHOT</version>
+        <version>0.3.1</version>
```
Comment (marcinczeczko) [inline knotx-core/knotx-server/pom.xml]: SNAPSHOT ?

File: knotx-core/knotx-template-engine/pom.xml [hunk 5]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>com.cognifide.knotx</groupId>
         <artifactId>knotx-core</artifactId>
-        <version>0.3.0-SNAPSHOT</version>
+        <version>0.3.1</version>
```
Comment (marcinczeczko) [inline knotx-core/knotx-template-engine/pom.xml]: SNAPSHOT ?

File: knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/TemplateEngineVerticle.java [hunk 6]
```diff
@@ -41,22 +43,26 @@
...
+
     @Override
     public void init(Vertx vertx, Context context) {
         super.init(vertx, context);
         JsonObject config = config().getJsonObject("config");
         this.serviceName = config.getString("service.name");
 
         configuration = new TemplateEngineConfiguration(config);
-        templateEngine = new TemplateEngine(this.vertx, configuration);
+        final HttpClientOptions httpClientOptions = new HttpClientOptions().setMaxPoolSize(1000).setKeepAlive(false);
```
Comment (marcinczeczko) [inline knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/TemplateEngineVerticle.java]: I think we should have this configurable on JSON level. Like for repository Verticle.

File: knotx-core/pom.xml [hunk 7]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>com.cognifide.knotx</groupId>
         <artifactId>knotx-root</artifactId>
-        <version>0.3.0-SNAPSHOT</version>
+        <version>0.3.1</version>
```
Comment (marcinczeczko) [inline knotx-core/pom.xml]: SNAPSHOT ?

File: knotx-example/knotx-example-monolith/pom.xml [hunk 8]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>com.cognifide.knotx</groupId>
         <artifactId>knotx-example</artifactId>
-        <version>0.3.0-SNAPSHOT</version>
+        <version>0.3.1</version>
```
Comment (marcinczeczko) [inline knotx-example/knotx-example-monolith/pom.xml]: SNAPSHOT ?

File: knotx-example/knotx-mocks/pom.xml [hunk 9]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>com.cognifide.knotx</groupId>
         <artifactId>knotx-example</artifactId>
-        <version>0.3.0-SNAPSHOT</version>
+        <version>0.3.1</version>
```
Comment (marcinczeczko) [inline knotx-example/knotx-mocks/pom.xml]: SNAPSHOT ?

File: knotx-example/pom.xml [hunk 10]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>com.cognifide.knotx</groupId>
         <artifactId>knotx-root</artifactId>
-        <version>0.3.0-SNAPSHOT</version>
+        <version>0.3.1</version>
```
Comment (marcinczeczko) [inline knotx-example/pom.xml]: SNAPSHOT ?

File: pom.xml [hunk 11]
```diff
@@ -24,7 +24,7 @@
 
     <groupId>com.cognifide.knotx</groupId>
     <artifactId>knotx-root</artifactId>
-    <version>0.3.0-SNAPSHOT</version>
+    <version>0.3.1</version>
```
Comment (marcinczeczko) [inline pom.xml]: SNAPSHOT ?

Comment (Skejven) [inline pom.xml]: No, this is ok. I will do 0.3.1 release and then go with 0.4.0-SNAPSHOT.

File: knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/TemplateEngineVerticle.java (hunk 6 above)
Comment (Skejven) [inline knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/TemplateEngineVerticle.java]: Added.

**Comment by Skejven** [comment]
Approved

**Pull request merged**